### PR TITLE
AP-2112 Bank account balance when there are no trasactions

### PR DESCRIPTION
**Pull request by stephenrichards**
## Bank account balance when there are no transactions

Normally, the balance reported on the means report is the latest balance from the bank account transactions.  However, if there are no transactions in the calculation period, then take the balance from the BankAccout record.

[Link to story](https://dsdmoj.atlassian.net/browse/AP-XXX)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests and rubocop should be passing: `bundle exec rake`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [ ] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [ ] The PR description should say what you changed and why, with a link to the JIRA story.
- [ ] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/models/legal_aid_application.rb [hunk 1]
```diff
@@ -176,6 +176,10 @@ def calculation_date
     LegalAidApplications::CalculationDateService.call(self)
   end
 
+  def year_to_calculation_date
+    [calculation_date - 1.year + 1.day, calculation_date]
```
Comment (Obsiye) [inline app/models/legal_aid_application.rb:180]: Hi, just wondering why `1.year`, i thought it'd be 3 months of transactions? Also how comes the `1.day` addition?

Comment (Obsiye) [reply app/models/legal_aid_application.rb:180]: ohh, is this related to "student loan" being annual? If so will that be the only type of irregular income (annual ones)?

**Review by Obsiye — APPROVED** [review]
LGTM! I left a question around the a `year_to_calculation_date`

File: app/models/legal_aid_application.rb (hunk 1 above)
Comment (stephenrichards) [reply app/models/legal_aid_application.rb:180]: yes - they declare the total amount they have received in the past year.  The means calculation works out what the monthly figure is, but on the certificate, we're just playing back to them what they have declared.  So the period is different (1 year as opposed to three months) .

**Pull request merged**